daemon/resize_test.go

```diff
@@ -0,0 +1,134 @@
+// +build linux
+
+package daemon
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/containerd/containerd"
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/daemon/exec"
+	"github.com/docker/docker/libcontainerd"
+	"github.com/gotestyourself/gotestyourself/assert"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// This test simply verify that when a wrong ID used, a specific error should be returned for exec resize.
+func TestExecResizeNoSuchExec(t *testing.T) {
+	n := "TestExecResize"
+	d := &Daemon{
+		execCommands: exec.NewStore(),
+	}
+	c := &container.Container{
+		ExecCommands: exec.NewStore(),
+	}
+	ec := &exec.Config{
+		ID: n,
+	}
+	d.registerExecCommand(c, ec)
+	err := d.ContainerExecResize("nil", 24, 8)
+	assert.ErrorContains(t, err, "No such exec instance")
+}
+
+type execResizeMockContainerdClient struct {
+	MockContainerdClient
+	ProcessID   string
+	ContainerID string
+	Width       int
+	Height      int
+}
+
+func (c *execResizeMockContainerdClient) ResizeTerminal(ctx context.Context, containerID, processID string, width, height int) error {
+	c.ProcessID = processID
+	c.ContainerID = containerID
+	c.Width = width
+	c.Height = height
+	return nil
+}
+
+// This test is to make sure that exec resize should acquire the lock inside exec-config object
+// If exec resize does not acquire the lock, there would be a race condition between exec start and resize.
+func TestExecResize(t *testing.T) {
```

Comment:
would a 10s timeout too long for a unit test?

Reply:
:man_shrugging:, I suppose yes. This is why I wanted it to be the request context :sweat_smile: 

/cc @thaJeztah do you mind the extra 10s until we pass down the REST context?

Reply:
Guess 10s is ok for now, if we have a big TODO on it, and an issue to track it so that we don't forget

Reply:
@mlaventure  @thaJeztah  I add a new unit test for resize without exec ready.  It would take 10s to finish the unittest.
> $ go test github.com/docker/docker/daemon -run ExecResize
> ok  	github.com/docker/docker/daemon	10.012s

Also, I have add "TODO" comment

Reply:
@zq-david-wang please also create an issue so that we don't forget about it :angel: :pray:

Reply:
@vdemeester copy that~!  
I will open an issue after the code merged.